crates/lib/tests/qvm_api.rs

```diff
@@ -6,28 +6,34 @@ use std::{collections::HashMap, num::NonZeroU16};
 use qcs::{
     client::Qcs,
     qvm::{
+        self,
         http::{self, HttpClient},
-        Client, QvmOptions,
+        QvmOptions,
     },
 };
 use regex::Regex;
 
 const PROGRAM: &str = r##"
 DECLARE ro BIT[2]
-H 0
-CNOT 0 1
+I 0
+X 1
 MEASURE 0 ro[0]
 MEASURE 1 ro[1]
 "##;
 
-async fn qvm_client() -> HttpClient {
+async fn http_qvm_client() -> HttpClient {
     let qcs_client = Qcs::load().await;
     HttpClient::from(&qcs_client)
 }
 
+fn libquil_qvm_client() -> qvm::libquil::Client {
+    qvm::libquil::Client {}
+}
+
+#[test_case::test_case(http_qvm_client().await ; "with HTTP client")]
+#[test_case::test_case(libquil_qvm_client() ; "with libquil client")]
 #[tokio::test]
-async fn test_get_version_info() {
-    let client = qvm_client().await;
+async fn test_get_version_info<C: qvm::Client>(client: C) {
```

Comment:
Can also do this, it amounts to the same thing:

```suggestion
async fn test_get_version_info(client: impl qvm::Client) {
```